packages/select/src/vaadin-select.js

```diff
@@ -627,6 +643,49 @@ class Select extends OverlayClassMixin(
     itemElement.setAttribute('id', this._itemId);
   }
 
+  /**
+   * @param {string} accessibleName
+   * @private
+   */
+  _accessibleNameChanged(accessibleName) {
+    this._srLabelController.setLabel(accessibleName);
+    this._setCustomAriaLabelledBy(accessibleName ? this._srLabelController.defaultId : null);
+  }
+
+  /**
+   * @param {string} accessibleNameRef
+   * @private
+   */
+  _accessibleNameRefChanged(accessibleNameRef) {
+    this._setCustomAriaLabelledBy(accessibleNameRef);
+  }
+
+  /**
+   * @param {string} ariaLabelledby
+   * @private
+   */
+  _setCustomAriaLabelledBy(ariaLabelledby) {
+    if (!this._items) {
+      return;
+    }
+    const selected = this._items[this._menuElement.selected];
+    const labelId = ariaLabelledby
+      ? `${ariaLabelledby} ${selected || this.placeholder ? this._itemId : ''}`.trim()
+      : null;
```

Comment:
This logic looks a bit complicated, consider extracting it to a separate method / getter.

Reply:
Refactored.